misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/FunctionMock.java

```diff
@@ -73,15 +73,6 @@ public static Object mockHandler(BObject mockFuncObj, Object... args) {
                 break;
             }
         }
-        if (returnVal == null) {
```

Comment:
Can we still detect if there is no return value registered with this change?

i.e. when the mock function is defined but `thenReturn`/`doNothing` is not specified within a test case?

Reply:
It cannot differentiate between the two scenarios. Will work on a fix for that 👍

Reply:
Updated the logic to differentiate two scenarios and updated tests to match each case.